### PR TITLE
add include_directories to use it in all subprojects

### PR DESCRIPTION
**Pull request by JoyIfBam5**
This is a fix for #244 
Frankly speaking I' not sure if it is a correct way to set the include directories for all sub-projects.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +114,8 @@ find_package(Boost REQUIRED)
 
 add_library(sml INTERFACE)
 
+include_directories(INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
```
Comment (gjasny) [inline CMakeLists.txt]: the right way to do it would be to use `target_link_libraries(myBenchmark PUBLIC sml)` for every target.

Comment (gjasny) [reply CMakeLists.txt]: I'd create a function that abstracts the following:
```
    add_executable(test_actions_defer actions_defer.cpp)
    target_link_libraries(test_actions_defer PUBLIC sml) 
    add_test(test_actions_defer test_actions_defer)
```

**Comment by krzysztof-jusiak** [comment]
Thanks @JoyIfBam5 and @gjasny :+1:

**Comment by JoyIfBam5** [comment]
I did not finish replacing all examples with a function. I will send another pull request as I will finish it.
My experience with git and pull requests is limited and I did not expect that my partial fix will join to this pull request.

**Comment by krzysztof-jusiak** [comment]
Sounds good @JoyIfBam5, thanks. Sorry for merging it too early, my bad :(

**Pull request merged**